cmd/query/app/server.go

```diff
@@ -131,7 +123,67 @@ func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.
 	healthServer.SetServingStatus("jaeger.api_v3.QueryService", grpc_health_v1.HealthCheckResponse_SERVING)
 
 	grpc_health_v1.RegisterHealthServer(server, healthServer)
-	return server, nil
+}
+
+func createGRPCServer(
+	ctx context.Context,
+	legacy bool,
+	querySvc *querysvc.QueryService,
+	metricsQuerySvc querysvc.MetricsQueryService,
+	options *QueryOptions,
+	tm *tenancy.Manager,
+	telset telemetery.Setting,
+) (*grpc.Server, error) {
+	var grpcServer *grpc.Server
+	if legacy {
+		var grpcOpts []grpc.ServerOption
+		if options.GRPC.TLSSetting != nil {
+			tlsCfg, err := options.GRPC.TLSSetting.LoadTLSConfig(ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			creds := credentials.NewTLS(tlsCfg)
+
+			grpcOpts = append(grpcOpts, grpc.Creds(creds))
+		}
+		if tm.Enabled {
+			grpcOpts = append(grpcOpts,
+				//nolint
+				grpc.StreamInterceptor(tenancy.NewGuardingStreamInterceptor(tm)),
+				grpc.UnaryInterceptor(tenancy.NewGuardingUnaryInterceptor(tm)),
+			)
+		}
+
+		grpcServer = grpc.NewServer(grpcOpts...)
+	} else {
+		var grpcOpts []configgrpc.ToServerOption
+		if tm.Enabled {
+			grpcOpts = append(grpcOpts,
+				//nolint
```

Comment:
?

Reply:
without this, the linter complains: `Function `NewGuardingStreamInterceptor->NewGuardingStreamInterceptor$1` should pass the context parameter (contextcheck)`

Reply:
can we do something like `//nolint: contextcheck` ?